x-pack/plugins/observability_solution/inventory/server/routes/entities/get_latest_entities.ts

```diff
@@ -62,17 +62,30 @@ export async function getLatestEntities({
 
   const query = [from, ...where, sort, limit].join(' | ');
 
-  const latestEntitiesEsqlResponse = await inventoryEsClient.esql<EntityInstance>(
+  const latestEntitiesEsqlResponse = await inventoryEsClient.esql<
+    {
+      'entity.id': string;
+      'entity.type': string;
+      'entity.definition_id': string;
+      'entity.display_name': string;
+      'entity.identity_fields': string | string[];
+      'entity.last_seen_timestamp': string;
+      'entity.definition_version': string;
+      'entity.schema_version': string;
+    },
+    { transform: 'unflatten' }
```

Comment:
We're unflattening the response 2 times. The first time by the `esql` function, and the second inside the `map` function. Could we change the response type to be like below?

```suggestion
  const latestEntitiesEsqlResponse = await inventoryEsClient.esql<
    EntityInstance,
    { transform: 'unflatten' }
```

And remove the call to unflattenObject inside the map?

Reply:
yes! thanks, missed it